linera-witty/tests/reentrancy.rs

```diff
@@ -0,0 +1,485 @@
+        <Caller::Runtime as Runtime>::Memory: RuntimeMemory<Caller>,
+    {
+        ImportedGetters::new(caller).get_s8()
+    }
+
+    fn get_u8<Caller>(caller: Caller) -> Result<u8, RuntimeError>
+    where
+        Caller: InstanceForImportedGetters,
+        <Caller::Runtime as Runtime>::Memory: RuntimeMemory<Caller>,
+    {
+        ImportedGetters::new(caller).get_u8()
+    }
+
+    fn get_s16<Caller>(caller: Caller) -> Result<i16, RuntimeError>
+    where
+        Caller: InstanceForImportedGetters,
+        <Caller::Runtime as Runtime>::Memory: RuntimeMemory<Caller>,
+    {
+        ImportedGetters::new(caller).get_s16()
+    }
+
+    fn get_u16<Caller>(caller: Caller) -> Result<u16, RuntimeError>
+    where
+        Caller: InstanceForImportedGetters,
+        <Caller::Runtime as Runtime>::Memory: RuntimeMemory<Caller>,
+    {
+        ImportedGetters::new(caller).get_u16()
+    }
+
+    fn get_s32<Caller>(caller: Caller) -> Result<i32, RuntimeError>
+    where
+        Caller: InstanceForImportedGetters,
+        <Caller::Runtime as Runtime>::Memory: RuntimeMemory<Caller>,
+    {
+        ImportedGetters::new(caller).get_s32()
+    }
+
+    fn get_u32<Caller>(caller: Caller) -> Result<u32, RuntimeError>
+    where
+        Caller: InstanceForImportedGetters,
+        <Caller::Runtime as Runtime>::Memory: RuntimeMemory<Caller>,
+    {
+        ImportedGetters::new(caller).get_u32()
+    }
+
+    fn get_s64<Caller>(caller: Caller) -> Result<i64, RuntimeError>
+    where
+        Caller: InstanceForImportedGetters,
+        <Caller::Runtime as Runtime>::Memory: RuntimeMemory<Caller>,
+    {
+        ImportedGetters::new(caller).get_s64()
+    }
+
+    fn get_u64<Caller>(caller: Caller) -> Result<u64, RuntimeError>
+    where
+        Caller: InstanceForImportedGetters,
+        <Caller::Runtime as Runtime>::Memory: RuntimeMemory<Caller>,
+    {
+        ImportedGetters::new(caller).get_u64()
+    }
+
+    fn get_float32<Caller>(caller: Caller) -> Result<f32, RuntimeError>
+    where
+        Caller: InstanceForImportedGetters,
+        <Caller::Runtime as Runtime>::Memory: RuntimeMemory<Caller>,
+    {
+        ImportedGetters::new(caller).get_float32()
+    }
+
+    fn get_float64<Caller>(caller: Caller) -> Result<f64, RuntimeError>
+    where
+        Caller: InstanceForImportedGetters,
+        <Caller::Runtime as Runtime>::Memory: RuntimeMemory<Caller>,
+    {
+        ImportedGetters::new(caller).get_float64()
+    }
+}
+
+/// Test reentrant functions with return values.
+///
+/// The host functions are called from the guest, and they return values obtained by calling back
+/// the guest through functions with the same names.
+#[test_case(MockInstanceFactory::default(); "with a mock instance")]
+#[cfg_attr(feature = "wasmer", test_case(WasmerInstanceFactory; "with Wasmer"))]
+#[cfg_attr(feature = "wasmtime", test_case(WasmtimeInstanceFactory; "with Wasmtime"))]
+fn getters<InstanceFactory>(mut factory: InstanceFactory)
```

Comment:
Can we start tests with `test_` ?

Reply:
I can do that, but I'm a bit hesitant to do so because the output will start looking like:

```
test test_getters::with_a_mock_instance ... ok
...
```

instead of just

```
test getters::with_a_mock_instance ... ok
...
```

What do you think?

Reply:
Well, the usual way to name functions is "do_something", and for tests, ideally "test_that_bla_does_this".
To me, "something" feels like a getter.

Reply:
For methods and normal functions I agree. But for test functions not so much, because the test functions aren't called by anybody explicitly.

Applied the suggestion because it doesn't matter much anyway :sweat_smile: